### PR TITLE
queryRange: Add more descriptive error messages

### PR DESCRIPTION
**Pull request by palash25**
Fixes: https://github.com/prometheus/prometheus/issues/4811

cc @codesome @gouthamve

### CODE REVIEW

File: web/api/v1/api.go [hunk 1]
```diff
@@ -310,10 +310,12 @@ func (api *API) query(r *http.Request) apiFuncResult {
 func (api *API) queryRange(r *http.Request) apiFuncResult {
 	start, err := parseTime(r.FormValue("start"))
 	if err != nil {
+		err = errors.New("missing parameter 'start'")
```
Comment (juliusv) [inline web/api/v1/api.go]: Currently this does `fmt.Errorf("cannot parse %q to a valid timestamp", s)` in `parseTime()`, so completely overwriting that error here would lose the actual timestamp value that couldn't be parsed. Also, the parameter is not technically missing, but invalid (can't be parsed). How about just:

`err = fmt.Errorf("invalid parameter 'start': %s", err)`

Similar for the instance below.

**Comment by juliusv** [comment]
Thanks! The DCO check is currently failing, could you follow https://github.com/prometheus/prometheus/pull/5229/checks?check_run_id=64963847 to add a signoff to your commit?

**Comment by palash25** [comment]
Thanks for the quick review @juliusv 

I have pushed the suggested changes with the commit signed-off

**Comment by juliusv** [comment]
@palash25 Thanks! Oh, one thing I overlooked yesterday is that the function above `queryRange` (`query`), also parses similar parameters and to be consistent we should make the error reporting the same there.

**Comment by palash25** [comment]
@juliusv requested changes made, please review

**Comment by juliusv** [comment]
:+1: Thanks!

**Pull request merged**